lib/KeyboardAwareHOC.js

```diff
@@ -587,7 +587,7 @@ function KeyboardAwareHOC(
     }
 
     _isCaretOutOfViewableArea(caretY: number, caretHeight: number, keyboardEndCoordinatesScreenY: number) {
-      return this._isCaretUnderKeyboard(caretY, caretHeight, keyboardEndCoordinatesScreenY) || 
+      return this._isCaretUnderKeyboard(caretY, caretHeight, keyboardEndCoordinatesScreenY - this.props.extraScrollHeight ) || 
```

Comment:
Consider the extra scroll height to calculate if the `caret is out of the view area`.

This is good to keep visible all the area we want to. But it might make the function name a bit confusing (this is no longer based on the caret alone).

Reply:
> This is good to keep visible all the area we want to

could you elaborate on this?

I couldn't understand why we need to change here.